src/core/common/timer.hpp

```diff
@@ -215,6 +215,44 @@ class TimerMilli : public Timer
     TimerMilliScheduler &GetTimerMilliScheduler(void) const;
 };
 
+/**
+ * This class implements a millisecond timer which also tracks an arbitrary context information.
+ *
+ * In typical `TimerMilli`/`TimerMicro` use, in the timer callback handler, the owner of the timer is determined using
+ * `GetOwner<Type>` method. This method works if there is a single instance of `Type` within OpenThread instance
+ * hierarchy. The `ContextTimerMill` is intended for cases where there may be multiple instances of the same class/type
+ * using a timer object. `ContextTimerMilli` will store a context `void *` information.
+ *
+ */
+class ContextTimerMilli : public TimerMilli
+{
+public:
+    /**
+     * This constructor creates a millisecond timer instance tracking an arbitrary context information.
```

Comment:
suggestion: "tracking an arbitrary context information" -> "that also maintains a user context pointer"